MetaMask/metamask-extension
# Swaps / STX improvements

## Explanation
This PR includes:
- Track real fees in ETH for approval and trade txs in the `Swap Completed` event
- Use USDConversionRate for calculating estimated fees in USD
- Fix a condition for checking non-prod METAMASK_ENVIRONMENT
- Load "parameters.stxMaxFeeMultiplier" from network config params, refactoring

## Manual Testing Steps

Track real fees in ETH for approval and trade txs in the `Swap Completed` event
- Check the Network tab after a swap is done and see that for the `Swap Completed` event we now track:
```
approval_gas_cost_in_eth: transactionsCost.approvalGasCostInEth,
trade_gas_cost_in_eth: transactionsCost.tradeGasCostInEth,
trade_and_approval_gas_cost_in_eth: transactionsCost.tradeAndApprovalGasCostInEth,
```

Use USDConversionRate for calculating estimated fees in USD
- In the `Swap Completed` event you can check that USD values are tracked correctly for these params:
```
reg_tx_fee_in_usd: undefined,
reg_tx_max_fee_in_usd: undefined,
stx_fee_in_usd: undefined,
stx_max_fee_in_usd: undefined,
```

Fix a condition for checking non-prod METAMASK_ENVIRONMENT
- When you run `yarn start` and select Rinkeby, Swaps are available now (they are disabled in production)

Load "parameters.stxMaxFeeMultiplier" from network config params, refactoring
- Check that on the View Quote page the max fee = 2 x estimated fee if Smart Transactions are enabled. The multiplier value (`2`) is loaded from backend and can be changed dynamically now
